frontends/concrete-python/examples/pir/PIR.ipynb

```diff
@@ -0,0 +1,509 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "id": "edb989d3-93b1-40cc-8ae4-5e739583c9bf",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Importing some libraries\n",
+    "\n",
+    "import argparse\n",
+    "import random\n",
+    "import time\n",
+    "\n",
+    "import numpy as np\n",
+    "\n",
+    "from concrete import fhe"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "af0a4710-ed46-4d21-a992-21eb2ad36bd2",
+   "metadata": {},
+   "source": [
+    "### One-hot vector\n",
+    "\n",
+    "Our database will be represented as a table T, and inputs to T will be `n_input_bits` bits. Cells of T are represented as `m_output_bits`.\n",
+    "\n",
+    "Before querying the database, the input `i` will be represented as one-hot vectors, ie a vector of `2**n_input_bits` bits: all of them are 0, but the one in the i-th position."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "id": "a1a50b5e-4c36-40e0-94b1-278fac66f322",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def make_one_hot_vector(index: int, index_max: int) -> np.array:\n",
+    "\n",
+    "    answer = np.zeros(shape=(index_max,), dtype=np.int8)\n",
+    "    answer[index] = 1\n",
+    "    return answer"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "5c265202-e5c4-4387-b67a-f9b8da1b2af8",
+   "metadata": {},
+   "source": [
+    "Here are a few examples of one-hot vectors."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "id": "861fb83d-783c-42e3-a59e-6abfeb274ff2",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "x = make_one_hot_vector(0, 5)\n",
+    "assert np.array_equal(x, np.array([1, 0, 0, 0, 0]))\n",
+    "\n",
+    "x = make_one_hot_vector(2, 7)\n",
+    "assert np.array_equal(x, np.array([0, 0, 1, 0, 0, 0, 0]))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "8a3d6bae-dac6-41d3-94fe-153906b99aad",
+   "metadata": {},
+   "source": [
+    "## Quering the database with a one-hot vector\n",
+    "\n",
+    "Quering the database will be very simple: the input will be given as a one-hot vector and is encrypted by the client. The database is in the clear on the server side. Just making a dot product with the database will return the right input."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "id": "87ea17f3-3c44-4f09-82d8-e5010f65bd8e",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "@fhe.compiler({\"one_hot_vector\": \"encrypted\", \"database\": \"clear\"})\n",
+    "def query(one_hot_vector: np.array, database: np.array) -> int:\n",
+    "    return np.dot(one_hot_vector, database)"
```

Comment:
I think `index_into_database` or `get_ith_element_of_database` would be a better name to describe the operation. `query` sounds like you want to compute `i in database` instead of `database[i]`.

Reply:
Renamed to get_ith_element_of_database